From 7c3a1e078b1131d027893ad1d0c37e93de3f541f Mon Sep 17 00:00:00 2001
From: andy3_kuo <andy3_kuo@asus.com>
Date: Mon, 20 Jan 2020 17:28:45 +0800
Subject: [PATCH 1/2] desktop-shell: Don't reposition views when output_list is
 empty

When all outputs have been destroyed, fetching the first output's
x, y, width and height would result in garbage values and in-turn would
result in wrong new-position values.
Instead, leave the view's position untouched so that when we have a new
output, it would be displayed in the same location.
---
 desktop-shell/shell.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index e5c3ef9e..cc38af1b 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -4752,6 +4752,9 @@ shell_reposition_view_on_output_destroy(struct weston_view *view)
 	float x, y;
 	int visible;
 
+	if (wl_list_empty(&ec->output_list))
+		return;
+
 	x = view->geometry.x;
 	y = view->geometry.y;
 
-- 
2.17.1

